lib/models/update-checker.js

```diff
@@ -69,24 +69,47 @@ UpdateChecker.prototype.checkNPM = function() {
   var concat = require('concat-stream');
 
   return new Promise(function(resolve, reject) {
-    http.get('http://registry.npmjs.org/ember-cli/latest', function(res) {
+
+    // default host and path, w/o proxy
+    var reqOptions = {
+      host: 'http://registry.npmjs.org',
+      path: '/ember-cli/latest'
+    };
+
+    if (this.settings.httpProxy){
+      var proxy = require('url').parse(this.settings.httpProxy);
+      reqOptions = {
+        host: proxy.hostname,
+        port: proxy.port,
+        path: 'http://registry.npmjs.org/ember-cli/latest'
+      };
+    }
+
+    http.get(reqOptions, function(res) {
       res.setEncoding('utf8');
 
       res.pipe(concat(function (data) {
         try {
+
           resolve(JSON.parse(data).version);
         } catch(error) {
           reject(error);
         }
       }));
-
       res.on('error', reject);
+    }).on('error', function(error){
+        reject(error);
     });
-  }).then(function(version) {
+
+  }.bind(this)).then(function(version) {
```

Comment:
this one? I am accessing this.settings.httpProxy, although it can be moved out where bind(this) will not be needed.